lib/composites/menu/events/main.js

```diff
@@ -17,29 +17,33 @@ module.exports = (elements) => {
    * Handle for clicks outside sideBar and dropdowns within topBar and close them if needed
    */
 
-  if (elements.menu && elements.trigger && elements.topBar) {
-    document.addEventListener('click', (e) => {
-      const target = e.target;
+  document.addEventListener('click', (e) => {
+    const target = e.target;
+    const topBar = closest(target, '.dqpl-top-bar li[data-dropdown-expanded]');
+
+    if (elements.menu && elements.trigger) {
       const isLocked = elements.menu.getAttribute('data-locked') === 'true';
       const isWithinSideBar = closest(target, '.dqpl-side-bar');
-      const isWithinDropdown = closest(target, '.dqpl-top-bar li[data-dropdown-expanded]');
       const isHamburger = Classlist(target).contains('dqpl-menu-trigger') ||
-        !!closest(target, '.dqpl-menu-trigger');
-      const expandedDropdownItem = document.querySelector('[data-dropdown-expanded="true"]');
+      !!closest(target, '.dqpl-menu-trigger');
       const menuExpanded = elements.menu.getAttribute('aria-expanded') === 'true';
 
-      if (!isWithinDropdown && expandedDropdownItem) {
-        expandedDropdownItem.click();
-      }
-
       if (!isWithinSideBar && !isHamburger && !isLocked && menuExpanded) {
         onTriggerClick();
       }
-    });
 
-    // hamburger clicks
-    elements.trigger.addEventListener('click', onTriggerClick);
-  }
+      // hamburger clicks
+      elements.trigger.addEventListener('click', onTriggerClick);
+    }
+
+    if (topBar) {
+      const expandedDropdownItem = document.querySelector('[data-dropdown-expanded="true"]');
+
+      if (!topBar && expandedDropdownItem) {
```

Comment:
huh? we have:

```js
if (topBar) {
  if (!topBar && somethingThatWillNeverBeChecked) {}
}
```

Reply:
I think you may be confused by what you've named stuff here.  `topBar` represents an `<li class="data-dropdown-expanded">` within a `dqpl-top-bar`.  Probably makes sense to name it more descriptively

Reply:
try something like this:
```js
const hasTopBarDropdown = closest(target, '.dqpl-top-bar li[data-dropdown-expanded]');

if (hasTopBarDropdown) {
  const expandedDropdownItem = document.querySelector('[data-dropdown-expanded="true"]');

  if (expandedDropdownItem) {
    expandedDropdownItem.click(); // toggle the menu closed
  }
}
```

Reply:
was totally confused. This makes sense now. 👍